daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -447,6 +448,9 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
     } yield ()
   }
 
+  private[engine] def enrich(typ: Type, value: Value[ContractId]): Result[Value[ContractId]] =
+    preprocessor.translateValue(typ, value).map(_.toValue)
```

Comment:
The implementation is a bit silly. 
Let's postpone a better if we need more efficiency.